### ISSUE TITLE
Can't use tabs in tests-names, info, etc...

### ISSUE THREAD

**Issue by nicolaje** [philsquared/Catch#476]
On Windows 7, with VS2013 Express edition, running a test such as this one:

```
TEST_CASE("A test case")
{
    INFO("info\twith\ttab");
    CHECK(false);
}
```

will crash at runtime with the message "string subscript out of range".

From what I understand, it hapens when Catch is constructing the messages to display in the console, but I don't know how/when.

![tabs](https://cloud.githubusercontent.com/assets/1287229/9135630/cb7a40c8-3d0f-11e5-885a-9247a3a5426b.png)

Thank you for your great work :+1:

**Comment by nicolaje** [comment]
Note this doesn't happen if the tab is in a string, such as in this situation:
```
TEST_CASE("A test case")
{
    string tab("\twith\ttab");
    INFO("info"<<tab);
    CHECK(false);
}
```

**Comment by jmmut** [comment]
I'm using the version 1.5.6 and there is still some problem. I'm using linux and it doesn't crash, but it does weird things.

For example, with the code of @nicolaje, the last `\t` disappears:
```
../test.cpp:128: FAILED:
  CHECK( false )
with message:
  info	withtab-
```

**Comment by lightmare** [comment]
Text formatting in Catch is buggy. Whether it crashes or not may depend on your compiler / stdlib / debug flags. This particular example doesn't crash for me, either (Linux); here's one that does:
```c++
INFO("0aaa_aaaa 1bbb_bbbb 2ccc_cccc 3ddd_dddd 4eee_eeee 5fff_ffff 6ggg_gggg 7hhh_hh\t  telltail");
CHECK(false);
```
I fixed this in #592, not sure whether that would also fix the crash on windows.

As for the disappearing `\t`, that's by design. Tabs are meant as indentation markers, they set the column where subsequent lines of the message (after splitting) start. Which means the first tab that wasn't removed might be considered a bug; but it's more a misuse; you're not supposed to construct long "staircase" messages with multiple tabs.

**Comment by philsquared** [comment]
I'm just completing a bit of a reworking of the text formatting code (which, frankly, was quite awful! I can only apologise for inflicting that code on the world).
In the process I'm trying to recall exactly what my original intention was with that tab handling. In Catch SelfTest, at least, it only seems to be used by the test that is specifically testing for it.
I don't recall ever documenting the behaviour - but it does seem to be causing issues here and there.
@lightmare you described the behaviour above. Did you reverse engineer that from the code, or have you seen it explained somewhere.

I'm strongly considering removing it entirely, since it seems to add unnecessary complexity to some already somewhat complex code.

**Comment by lightmare** [comment]
@philsquared My description above was inferred from the code, while I was trying to fix the out of range error. The way `Text` constructor works is it finds the last tab in the first `width` characters, remembers the position and deletes this tab -- that's why I think it was meant as a mere marker; whether leaving all preceding tab characters in place was intentional or not, I can't say.

**Comment by philsquared** [comment]
@lightmare thanks for getting back.
Yes, that's *what* it does. My question was just whether your knowledge of that was just from looking at the code - and you've answered that - thanks.
While it seems like a handy feature I don't think it's ever been used - and maybe is overkill for use in a test framework. So I'm leaning towards removing it (doing so fixes a few issues - including this one, I believe - although my rewrite fixes more - hopefully all).

**Comment by lightmare** [comment]
It's a perfect feature for `--help`, though, maybe that's what you intended it for.
See [Boost.Program_options](http://www.boost.org/doc/libs/1_60_0/doc/html/program_options/overview.html#idm45555218362096). They also find a tab, remember position and delete it. Multiple tabs cause error.

**Comment by philsquared** [comment]
Ah yes! I used to have a much richer help system built in - that's what it was for!
For now it's not being used for that (in Catch). It may make sense to restore it in Clara (at this point I'm not sure if I'm going to continue the C++03 version of Clara - or just keep that forked in Catch Classic and have a C++11 Clara be part of Catch2)

**Comment by horenmar** [comment]
This has been fixed in 4a04682e491b5697979e921c3a5ed9ae933d9406, but for time being the `\t` special handling was removed. I am going to close this with a "revisit" label in case we want to restore the feature.
